test/mock/grpc/mock_management_plane_grpc_server.go

```diff
@@ -44,9 +51,23 @@ func NewManagementGrpcServer() *ManagementGrpcServer {
 		if mgs.connectionRequest == nil {
 			c.JSON(http.StatusNotFound, nil)
 		} else {
+			mgs.connectionMutex.Lock()
 			c.JSON(http.StatusOK, gin.H{
 				"connectionRequest": mgs.connectionRequest,
 			})
+			mgs.connectionMutex.Unlock()
+		}
+	})
+
+	server.GET("/api/v1/status", func(c *gin.Context) {
+		if mgs.updateDataPlaneStatusRequest == nil {
+			c.JSON(http.StatusNotFound, nil)
+		} else {
+			mgs.updateDataPlaneStatusMutex.Lock()
+			c.JSON(http.StatusOK, gin.H{
+				"updateDataPlaneStatusRequest": mgs.updateDataPlaneStatusRequest,
+			})
+			mgs.updateDataPlaneStatusMutex.Unlock()
 		}
```

Comment:
If we are gonna lock for responses, should we also lock for the first branch of the if-clause?
```suggestion
        mgs.updateDataPlaneStatusMutex.Lock()
		if mgs.updateDataPlaneStatusRequest == nil {
			c.JSON(http.StatusNotFound, nil)
		} else {
			c.JSON(http.StatusOK, gin.H{
				"updateDataPlaneStatusRequest": mgs.updateDataPlaneStatusRequest,
			})
		}
		mgs.updateDataPlaneStatusMutex.Unlock()
```